src/OpenTelemetry/Metrics/AggregatorStore.cs

```diff
@@ -178,27 +179,46 @@ private void InitializeZeroTagPointIfNotInitialized()
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private int LookupAggregatorStore(string[] tagKeys, object[] tagValues, int length)
         {
-            int aggregatorIndex;
-            string[] seqKey = null;
+            var givenTags = new Tags(tagKeys, tagValues);
 
-            // GetOrAdd by TagKeys at 1st Level of 2-level dictionary structure.
-            // Get back a Dictionary of [ Values x Metrics[] ].
-            if (!this.keyValue2MetricAggs.TryGetValue(tagKeys, out var value2metrics))
+            // We only need to sort if there is more than one Tag Key.
+            if (!this.sortedTagsDictionary.TryGetValue(givenTags, out var sortedTags))
             {
                 // Note: We are using storage from ThreadStatic, so need to make a deep copy for Dictionary storage.
```

Comment:
Just for my learning, I assume that we are making a deep copy here because this is happening in the main thread (thread A), and we want to achieve parallelism (non-blocking) call in another thread (thread B). 
We want to make sure the if thread A is modifying tags, thread B won't got affected. Is my assumption correct?
If that is the case, could you point me to where is the logic of thread A spawning thread B? Thanks!

Reply:
Or maybe, it is because the users can spawn more than 1 thread when doing their instrumentation, so it is not necessarily that main thread is spawning a child thread...

Reply:
We need to store the tagKeys and tagValues in the dictionary. The tagKeys and tagValues provided to the method `LookupAggregatorStore` are local to the thread that is running that code. These values will keep changing as long as different metrics are emitted on that thread. The dictionary entries should not be modified once they are in the dictionary. Hence, we need an allocated memory space to store these. That's why we do a deep copy.